File: insfrastructure/usecases/update_account_usecase.py
from datetime import datetime
from insfrastructure.entities.bank_account_entity import BankAccountEntity
from insfrastructure.repositories.bank_account_repository import BankAccountRepository


class UpdateAccountUseCase:
    def __init__(self, bank_account_repo: BankAccountRepository):
        self.bank_account_repo = bank_account_repo

    def execute(self, bank_account_id: int, first_name: str, last_name: str, date_of_birth: datetime, email: str, national_id: str, phone_number: str, parmanent_address: str) -> bool:
        return self.bank_account_repo.update(bank_account_id, first_name, last_name, date_of_birth, email, national_id, phone_number, parmanent_address)

File: typescript-impl/src/infrastructure/usecases/update_account_usecase.ts
import { BankAccountRepository } from '../repositories/bank_account_repository';

export class UpdateAccountUseCase {
    private bank_account_repo: BankAccountRepository;

    constructor(bank_account_repo: BankAccountRepository) {
        this.bank_account_repo = bank_account_repo;
    }

    public execute(bank_account_id: number, first_name: string, last_name: string, date_of_birth: Date, email: string, national_id: string, phone_number: string, parmanent_address: string): boolean {
        return this.bank_account_repo.update(bank_account_id, first_name, last_name, date_of_birth, email, national_id, phone_number, parmanent_address);
    }
}
